packages/aquery/src/utils/form/datePicker.js

```diff
@@ -3,39 +3,71 @@ import randomId from '../randomId';
 const FORMATS = { FORMAT_US: 'MM-DD-YYYY', FORMAT_WORLD: 'DD-MM-YYYY' };
 
 const addDatePicker = (date, $) => {
-  if (date) {
-    const input = date.input;
-    if (!input.getAttribute('name')) {
-      input.setAttribute('name', randomId());
-    }
-    const datePicker = document.createElement('amp-date-picker');
-    datePicker.setAttribute(
-      'input-selector',
-      `[name=${input.getAttribute('name')}]`,
-    );
-    if (date.mode && date.mode === 'static') {
-    } else {
-      datePicker.setAttribute('mode', 'overlay');
-      datePicker.setAttribute('layout', 'container');
-    }
-    const format = resolveFormat(date.format);
-    datePicker.setAttribute('format', format);
-    if (date.required) {
-      input.removeAttribute('readonly');
-      input.setAttribute('required', '');
-      datePicker.setAttribute('required', '');
-      console.log('addDatePicker ', datePicker);
+    if (date) {
+        let mode_pre
+        date.mode ? mode_pre = date.mode : mode_pre = 'lightbox';
+        const mode = mode_pre;
+        const input = date.input;
+        if (!input.getAttribute('name')) {
+            input.setAttribute('name', randomId());
+        }
+        const datePicker = document.createElement('amp-date-picker');
+        datePicker.setAttribute(
+            'input-selector',
+            `[name=${input.getAttribute('name')}]`
+        );
+        const format = resolveFormat(date.format);
+        datePicker.setAttribute('format', format);
+
+        if (date.required) {
+            input.removeAttribute('readonly');
+            input.setAttribute('required', '');
+            datePicker.setAttribute('required', '');
+            console.log('addDatePicker', datePicker);
+        }
+
+        if (date.mode && date.mode === 'static') {} else if (mode === "overlay") {
+            datePicker.setAttribute('mode', 'overlay');
+            datePicker.setAttribute('layout', 'container');
+            const parent = input.parentNode;
+            parent.appendChild(datePicker);
+            datePicker.appendChild(input);
+        } else {
+            var lightboxID = `lb${input.getAttribute('name')}`;
```

Comment:
Use const

Reply:
but i need to use it outside the if. Will it work there if it's const?

Reply:
If you needed to use it outside of the scope of the if, var wouldn't help either...

You would have needed to declare the variable outside of the scope, and then initialize it inside the scope. If you needed to do that - it should have been "let" as using var is deprecated.

Anyhow, in this case it's a const.